reference-implementation/lib/exclusive-stream-reader.js

```diff
@@ -89,11 +79,11 @@ export default class ExclusiveStreamReader {
     }
 
     if (this._encapsulatedReadableStream._readableStreamReader !== this) {
-      return this._closedAfterRelease;
+      return this._closedPromise;
     }
 
-    this.releaseLock();
-    return this._encapsulatedReadableStream.cancel(reason, ...args);
+    // Bypass lock check.
+    return CancelReadableStream(this._encapsulatedReadableStream, reason, ...args);
```

Comment:
This will effect: spec's class definition block; spec header; spec step 3; and CancelReadableStream definition. They should all just take a _reason_ argument.